src/storage/src/hummock/store/version.rs

```diff
@@ -551,29 +537,38 @@ pub fn read_filter_for_batch(
         }));
     }
 
-    Ok((imm_vec, sst_vec, table_watermark))
+    Ok((
+        key_range,
+        (imm_vec, sst_vec, max_mce_version, table_watermark),
+    ))
 }
 
 pub fn read_filter_for_local(
     epoch: HummockEpoch,
     table_id: TableId,
-    table_key_range: &TableKeyRange,
+    mut table_key_range: TableKeyRange,
     read_version: Arc<RwLock<HummockReadVersion>>,
-) -> StorageResult<ReadVersionTuple> {
+) -> StorageResult<(TableKeyRange, ReadVersionTuple)> {
     let read_version_guard = read_version.read();
+
+    let committed_version = read_version_guard.committed().clone();
+
+    let table_watermark = read_version_guard
```

Comment:
Is it possible to get the "rewrite" key_range before calling `read_filter_for_***` instead of inside to avoid returning a `rewrite` key_range?

Reply:
For `read_filter_for_local` we can do it but for `read_filter_for_batch` we cannot, because we may have multiple read versions, and if we want to rewrite the key range before calling it, we need to acquire the read lock for twice, once for rewrite the key range, and once for later pruning the imm and sst.

So currently I think we'd better align the usage of `read_filter_for_***`.